### ISSUE TITLE
Imports accessed by Hardhat before preprocessing

### ISSUE THREAD

**Issue by ItsNickBarry** [wighawag/hardhat-preprocessor#3]
In a [previous issue](https://github.com/wighawag/buidler-preprocessor/issues/4) I reported that `import` statements were processed incorrectly.

It seems that while Hardhat handles imports slightly differently, preprocessing of imports still does not work.  For example:

```solidity
// Test.sol

import 'asdf';
```

```javascript
// hardhat.config.js

// ...

module.exports = {
  preprocess: {
    eachLine: () => ({ transform: (line) => line.replace('asdf', 'ghjk') }),
  },
};
```

This code results in the following error:

```
Error HH411: The library asd, imported from Test.sol, is not installed. Try installing it using npm.
```

**Comment by wighawag** [comment]
I see, the issue is that hardhat imports are parsed first before being passed to the task that hardhat-preprocessor override: TASK_COMPILE_SOLIDITY_GET_COMPILATION_JOB_FOR_FILE

I could potentially override TASK_COMPILE_SOLIDITY_GET_DEPENDENCY_GRAPH but the typing indicate that this is not expected (ResolvedFile has readonly field) 

I ll investigate further but this might require a change in hardhat. It would be great if hardhat offer a filesystem access hook of some sort so preprocessor can simply transform the file in memory at the lowest level possible, @alcuadrado ?

**Comment by ItsNickBarry** [comment]
I don't particularly need this feature, so it might be best to wait for a greater need before putting effort into it.

**Comment by alcuadrado** [comment]
Hey @wighawag, I've just seen this. Try overriding this task instead: https://github.com/nomiclabs/hardhat/blob/master/packages/hardhat-core/src/builtin-tasks/compile.ts#L141

**Comment by wighawag** [comment]
Thanks @alcuadrado that's perfect, did not know it was there.

@ItsNickBarry it is now implemented in 0.1.2 but it require at least hardhat 2.0.5
